cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/prolog

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+exec &>>"/var/log/parallelcluster/slurm_prolog_epilog.log"
+timestamp=`date +"%Y-%m-%d %H:%M:%S,%3N"`
+echo "$timestamp Adding nodes for job ${SLURM_JOB_ID} to /etc/hosts"
+myip=$(hostname -I)
+added=false   # If the nodes have been added to /etc/hosts
+no_add=false   # If it is not a safe line in /etc/hosts to add the nodes
+HOSTS=`echo "$SLURM_NODE_ALIASES" |awk 'BEGIN{RS=","; FS=":"}; {gsub(/\[|\]/,"",$2); print $2,$1,$3}'`
+# Read /etc/hosts line by line and output the lines to /etc/tmphosts. Then overwrite /etc/hosts with /etc/tmphosts
+test -f /etc/tmphosts && rm /etc/tmphosts
+while IFS= read -r line
```

Comment:
this looks a pretty complex logic. have you considered using sed to insert before the first occurrence of #HOSTS?

Reply:
Good idea. I was thinking about inserting after ip, which led me to this complex logic.
Inserting before #HOSTS should be easier

Reply:
Done